pkg/server/cleaner.go

```diff
@@ -0,0 +1,83 @@
+package server
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/gocardless/draupnir/pkg/exec"
+	"github.com/gocardless/draupnir/pkg/models"
+	"github.com/gocardless/draupnir/pkg/store"
+	"github.com/prometheus/common/log"
+	google "google.golang.org/api/oauth2/v1"
+)
+
+type InstanceCleaner struct {
+	logger        log.Logger
+	instanceStore store.InstanceStore
+	executor      exec.Executor
+	service       googleService
+}
+
+type googleService interface {
+	Tokeninfo() *google.TokeninfoCall
+}
+
+func NewInstanceCleaner(logger log.Logger, instanceStore store.InstanceStore, executor exec.Executor, service googleService) *InstanceCleaner {
+	return &InstanceCleaner{
+		logger:        logger,
+		instanceStore: instanceStore,
+		executor:      executor,
+		service:       service,
+	}
+}
+
+func (ic *InstanceCleaner) Start(cctx context.Context, interval time.Duration) error {
+	for {
+		select {
+		case <-time.After(interval):
+			ic.logger.Info("cleaning old instances with invalid tokens")
+			instances, err := ic.instanceStore.List()
+			if err != nil {
+				ic.logger.Info("skipping cleaning old instances - could not get instances list")
```

Comment:
Oh a sentry I think would be good for this, and an error log.

I was thinking that this would more likely be a transient network error as opposed to the DB being totally down. We don't have any observability of draupnir but it we did we would know about this and would be able to handle it better.